components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -183,15 +198,34 @@ public BrandingPreference resolveBrandingPreference(String type, String name, St
         return getBrandingPreference(type, name, locale);
     }
 
+    @Override
+    public BrandingPreference resolveApplicationBrandingPreference(String name, String locale)
+            throws BrandingPreferenceMgtException {
+
+        try {
+            return getBrandingPreference(APPLICATION_TYPE, name, locale);
+        } catch (BrandingPreferenceMgtClientException e) {
+            if (ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Can not find a branding preference configurations for Application: " + name);
+                }
+                // Return default branding preference.(organization level branding preferences)
+                return resolveBrandingPreference(ORGANIZATION_TYPE, name, locale);
```

Comment:
Check if we can get rid of the name parameter. Or else get the tenant domain before this and pass that as the name for org level.

Reply:
Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/13/commits/cf770738c9203eb8c32f84bd8258e30b6acf198a